### PR TITLE
pds api 127: improved error messaging

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Changed all of the AcceptTypeException to return 406 instead of 501.

Improved the message for the exception to include all of the known Accept: content types allowed.

Note, if a Accept: content type is not supported, then the result always comes out as JSON. If the request was 'Accept: text/htlm' then the chosen serializer is JSON.

## ⚙️ Test Data and/or Report
```
$ curl --location --request GET 'http://localhost:8080/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_ins::1.0' --header 'Accept: text/html'
<html><body><h1>Error Message</h1><h2>From Request</h2><p>/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_ins::1.0</p><h2>Message</h2><p>The lidvid urn:nasa:pds:izenberg_pdart14_meap:document:ns_ins was not found</p></body></html>
```

```
$ curl --location --request GET 'http://localhost:8080/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst::1.0' --header 'Accept: text/htmls'
{"request":"/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst::1.0","message":"The given application type, text/htmls, is not known by RquestAndResponseContext. Known choices are: application/csv,application/xml,text/html,application/json,*/*,application/pds4+xml,text/csv,text/xml,application/kvp+json,application/pds4+json"}
```

## ♻️ Related Issues

NASA-PDS/registry-api#446

### CODE REVIEW

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

Ready for review.

File: service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java [hunk 1]
```diff
@@ -161,10 +161,17 @@ private RequestAndResponseContext(
 		}
 		else
 		{
+			String known = "";
 			log.warn("Could not find a matach for application type: " + String.valueOf(this.format));
 			log.warn("   Known types: " + String.valueOf(this.formatters.keySet().size()));
-			for (String key : this.formatters.keySet()) log.warn("      key: " + String.valueOf(key));
-			throw new ApplicationTypeException("The given application type, " + String.valueOf(this.format) + ", is not known by RquestAndResponseContext.");
+			for (String key : this.formatters.keySet())
```
Comment (tloubrieu-jpl) [inline service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java]: Maybe you could use a method like 

String known = String.join(",", this.formatters.keySet());

That would be shorter and I feel like that would be more understandable as well but I did not test it.

File: service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java [hunk 2]
```diff
@@ -161,10 +161,17 @@ private RequestAndResponseContext(
...
+			{
+				log.warn("      key: " + String.valueOf(key));
```
Comment (tloubrieu-jpl) [inline service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java]: Do we need a warning for this message ? I would think of DEBUG, unless I misunderstood what it does.

File: service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java [hunk 3]
```diff
@@ -161,10 +161,17 @@ private RequestAndResponseContext(
...
+				known = known + key + ",";
+			}
+			known = known.substring(0, known.length()-1);
+			throw new ApplicationTypeException("The given application type, " + String.valueOf(this.format) +
```
Comment (tloubrieu-jpl) [inline service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java]: I would rephrase the message like:

"The Accept header value " + String.valueOf(this.format) + " is not supported, supported values are " + known

Maybe that sentence does not make sense in the context of the exception but we need a message which makes sense to the user of the API, so that he understands what he need to change in his request.

**Review by tloubrieu-jpl — CHANGES_REQUESTED** [review]
Thanks @al-niessner , that works well. the most critical comment is how the message shows to the user.

File: service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java (hunk 2 above)
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java]: Need in what way? Will the program work without the warning, yes. When a user says they got a 406 on Saturday will you be able to find it in the log if not posted, no. So define what you mean by need.

Warnings are usually issued when the program detects something fishy (a handled exception) notifying the initiator of the fishy bit that something looks wrong but was handled albeit maybe not how the initiator wanted so they may want to try again -- think or google compiler warnings. Warnings differs from errors in that the program does not crash and burn. Debug on the other hand is way of tracing the program to determine where a bug is hiding or may exist or prove its existance.

For this message when a user mentions days later that a 406 was returned (something fishy but handled) we would have the appropriate level of logging.

If you want it to be debug then simply say so directly and I will change it. One could argue that given errors are being returned to the user the logging is superfluous.

File: service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java (hunk 3 above)
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java]: resolved

File: service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java (hunk 1 above)
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java]: resolved

**Review by tloubrieu-jpl — APPROVED** [review]
Thanks Al, I will merge it.
